Coroutines home work

Mr-Ratatu opened the pull request.

Mr-Ratatu committed:
Coroutines home work

ZadorozhnyiSemen commented on app/src/main/java/otus/homework/coroutines/ExceptionWrapper.kt:
```diff
@@ -0,0 +1,40 @@
+    ) {
+        try {
+            onSuccess.invoke()
+        } catch (error: Exception) {
```
Порядок типов влияет на обработку ошибок, в нашем случае, если выпадет SocketTimeoutException то он обработается в этом catch, потому что SocketTimeoutException наследуется от Exception и мы не попадем в catch ниже, нужно поменять их местами https://tinyl.io/5n6K

Mr-Ratatu replied:
Точно, как-то не подумал об этом) спасибо, буду иметь ввиду

ZadorozhnyiSemen commented on app/src/main/java/otus/homework/coroutines/presentation/CatsViewModel.kt:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    fun onInitComplete() {
+        viewModelScope.launch(exceptionHandler { error ->
```
Не хватает логгирования в CrashMonitor как это сделано в методе apiCall

Mr-Ratatu replied:
Добавил

ZadorozhnyiSemen reviewed (changes requested):
Спасибо, работа хорошая, есть два небольших замечания, после которых все условия будут выполнены

Mr-Ratatu committed:
Correction of omissions
- changed position for catch blocks
- added call CrashMonitor in exceptionHandler method
